code/mob/living/critter/flockcritter_parent.dm

```diff
@@ -354,46 +354,59 @@
 
 	onStart()
 		..()
-		if(target)
-			var/mob/living/critter/flock/F = owner
-			var/T = target
-			var/mob/living/critter/flock/C
-			if(istype(T, /mob/living/critter/flock))
-				C = T
-			if(F)
-				if(C)
-					F.tri_message("<span class='notice'>[owner] begins spraying glowing fibres onto [C].</span>",
-						F, "<span class='notice'>You begin repairing [C.real_name]. You will both need to stay still for this to work.</span>",
-						T, "<span class='notice'>[F.real_name] begins repairing you. You will both need to stay still for this to work.</span>",
-						"You hear hissing and spraying.")
-				else
-					F.tri_message("<span class='notice'>[owner] begins spraying glowing fibres onto [T].</span>",
-						F, "<span class='notice'>You begin repairing [T]. You will both need to stay still for this to work.</span>",
-						T, "<span class='notice'>[F.real_name] begins repairing you. You will both need to stay still for this to work.</span>",
-						"You hear hissing and spraying.")
-				playsound(T, "sound/misc/flockmind/flockdrone_quickbuild.ogg", 50, 1)
-				if(C?.is_npc)
-					C.ai.wait()
+		if(!target || !owner)
+			return
+		var/mob/living/critter/flock/F = owner
+		var/mob/living/critter/flock/C
+		if(istype(target, /mob/living/critter/flock))
+			C = target
+		if(C)
+			F.tri_message("<span class='notice'>[owner] begins spraying glowing fibers onto [C].</span>",
+				F, "<span class='notice'>You begin repairing [C.real_name]. You will both need to stay still for this to work.</span>",
+				target, "<span class='notice'>[F.real_name] begins repairing you. You will both need to stay still for this to work.</span>",
+				"You hear hissing and spraying.")
+			if (C.is_npc)
+				C.ai.wait()
+		else
+			F.tri_message("<span class='notice'>[owner] begins spraying glowing fibers onto [target].</span>",
+				F, "<span class='notice'>You begin repairing [target]. You will need to stay still for this to work.</span>",
+				null, null, "You hear hissing and spraying.")
+		playsound(target, "sound/misc/flockmind/flockdrone_quickbuild.ogg", 50, 1)
 
 	onEnd()
 		..()
+		if (!owner)
+			return
+		if (istype(target, /mob/living/critter/flock))
+			var/mob/living/critter/flock/flockcritter = target
+			flockcritter.HealDamage("All", flockcritter.health_brute / 3, flockcritter.health_burn / 3)
+			if (flockcritter.is_npc)
+				flockcritter.ai.interrupt()
+		else if (istype(target, /obj/flock_structure))
+			var/obj/flock_structure/structure = target
+			structure.repair()
+		else
+			switch (target.type)
+				if (/obj/machinery/door/feather)
```

Comment:
This is kinda cursed, but the only way to fix it would be to componentize repair or use the crime `:` operator so it's probably fine

Reply:
A whole lot in flock can be componentized :shelterfrog;